lib/ng2-select2.component.ts

```diff
@@ -95,13 +112,37 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
         }
 
         this.element.on('select2:select select2:unselect', () => {
+            this.onChange(that.element.val());
+            this.onTouched();
             this.valueChanged.emit({
-                value: this.element.val(),
-                data: this.element.select2('data')
```

Comment:
The whole record as e.g. returned via AJAX. If my AJAX returns something like this:

```
{ id: 1, text: 'foo', extraField: true, anotherField: 'bar' }
```
Then this record would be returned by `this.element.select2('data')`.